tf2/include/tf2/buffer_core.h

```diff
@@ -108,7 +109,7 @@ class BufferCore
 
   /** \brief Clear all data */
   TF2_PUBLIC
-  void clear();
+  void clear() override;
```

Comment:
I'm also mildly concerned about making these functions virtual (though I don't see an alternative), because it will now require a vtable lookup, and these functions are called very frequently. Just bringing it up in case anyone else agrees, but I don't see an alternative at the moment as the standard way to avoid a vtable is CRTP and the way we're using this with plugins I think makes that option not viable.